packages/grafana-runtime/src/utils/DataSourceWithBackend.ts

```diff
@@ -25,17 +25,31 @@ export class DataSourceWithBackend<
    * Ideally final -- any other implementation would be wrong!
    */
   query(request: DataQueryRequest): Observable<DataQueryResponse> {
-    const { targets, intervalMs, maxDataPoints, range } = request;
+    const { targets, intervalMs, maxDataPoints, range, requestId } = request;
+    let rangeFrom: any = undefined;
+    let rangeTo: any = undefined;
+
+    if (range) {
+      rangeFrom = range.from.valueOf().toString();
+      rangeTo = range.to.valueOf().toString();
+    }
 
     let expressionCount = 0;
-    const orgId = config.bootData.user.orgId;
     const queries = targets.map(q => {
       if (q.datasource === ExpressionDatasourceID) {
         expressionCount++;
         return {
-          ...q,
           datasourceId: this.id,
-          orgId,
+          datasourceName: q.datasource,
+          refId: q.refId,
+          from: rangeFrom,
+          to: rangeTo,
+          intervalMs,
+          maxDataPoints,
+          model: {
+            ...q,
+            datasourceId: this.id,
```

Comment:
does this need `datasourceId`?  duplicated in the object above?

Reply:
Probably not, but think GEL is dependent on it. Will remove

Reply:
now that GEL has its own endpoint... i assume it can do whatever it wants with the parent element info too